### ISSUE TITLE
reach router navigate causes 404 

### ISSUE THREAD

**Issue by zebapy** [gatsbyjs/gatsby#7643]
## Description

Using a MobileNav component like so

```jsx
import React, { Component } from 'react';
import { navigate } from '@reach/router';

class MobileNav extends Component {
  handleChange = event => {
    navigate(event.target.value);
  };
  render() {
    const { items } = this.props;

    return (
      <div className="mobile docs-nav">
        <select className="btn-primary" onChange={this.handleChange}>
          <option>Select A Topic</option>
          {items.map(item => (
            <optgroup label={item.title} key={item.title}>
              {item.group.edges.map(({ node }) => (
                <option value={node.fields.slug} key={node.fields.slug} className="nav-link">
                  {node.frontmatter.title}
                </option>
              ))}
            </optgroup>
          ))}
        </select>
      </div>
    );
  }
}

export default MobileNav;
```

When user changes the `select`, the page becomes a 404, though if I refresh, the page works fine. 

### Steps to reproduce

1. clone https://github.com/zebapy/netlify-cms/tree/gatsby-v2
2. `cd website && yarn && yarn start`
3. go to /docs 
4. resize viewport smaller so you get MobileNav to show
5. Change the MobileNav select input to another page

### Expected result
Page should transition to the selected item from mobile nav

### Actual result
Page is 404

### Environment

  System:
    OS: OS X El Capitan 10.11.6
    CPU: x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
    Shell: 5.0.8 - /bin/zsh
  Binaries:
    Node: 8.10.0 - /usr/local/bin/node
    Yarn: 1.7.0 - /usr/local/bin/yarn
    npm: 6.2.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 68.0.3440.106
    Firefox: 61.0.1
    Safari: 11.1.2
  npmPackages:
    gatsby: next => 2.0.0-beta.105 
    gatsby-plugin-catch-links: next => 2.0.2-beta.9 
    gatsby-plugin-manifest: next => 2.0.2-beta.6 
    gatsby-plugin-postcss: ^1.0.0 => 1.0.0 
    gatsby-plugin-react-helmet: next => 3.0.0-beta.4 
    gatsby-remark-autolink-headers: next => 2.0.0-beta.5 
    gatsby-remark-prismjs: next => 3.0.0-beta.7 
    gatsby-source-filesystem: next => 2.0.1-beta.10 
    gatsby-transformer-json: next => 2.1.1-beta.5 
    gatsby-transformer-remark: next => 2.1.1-beta.6 
    gatsby-transformer-yaml: next => 2.1.1-beta.3 
  npmGlobalPackages:
    gatsby-cli: 1.1.25

**Comment by zebapy** [comment]
@Chuloo does the reproduction steps not work or does this status mean someone needs to test it?

**Comment by momo-sa** [comment]
> You can't use @reach/router's navigate as it doesn't wait for the page resources to load which means like you found @kylegalbraith, the page errors (as the code is missing). Follow this guide https://next.gatsbyjs.org/docs/gatsby-link/#programmatic-navigation

Makes sense. To summarize for future readers and check my own understanding: `@reach/router`'s `navigate` can be useful for something like modifying query params, granting that the resources are loaded, but not linking to other pages. Instead we can use Gatsby's own `navigate`, which as far as I've messed with appears to work just fine in both cases. https://www.gatsbyjs.com/docs/gatsby-link/#how-to-use-the-navigate-helper-function

Someone looking for a quick solution without changing their approach can try `import { navigate } from "gatsby"` instead of from `@reach/router`.
